src/rollup/widgets/init-widget.js

```diff
@@ -58,85 +69,79 @@ export default function initializeWidget (widget) {
     }
   }
 
-  var fields = ['msg', 'headline', 'image', 'confirmAction.callback'];
+  // entity fields
+  widget.valid = widget.valid && entityFieldChecker(widget, pf.customData);
 
-  pf.entityFieldChecker(fields, widget, function () {
-    // display conditions based on page load
-    if (condition.date) {
-      widget.valid = widget.valid && dateChecker(condition.date);
-    }
+  // display conditions based on page load
+  if (condition.date) {
+    widget.valid = widget.valid && dateChecker(condition.date);
+  }
 
-    if (condition.pageVisits) {
-      widget.valid = widget.valid && pageVisitsChecker(condition.pageVisits);
-    }
+  if (condition.pageVisits) {
+    widget.valid = widget.valid && pageVisitsChecker(condition.pageVisits);
+  }
 
-    if (condition.hideAfterAction) {
-      widget.valid =
-        widget.valid &&
-        hideAfterActionChecker(condition.hideAfterAction, widget);
-    }
+  if (condition.hideAfterAction) {
+    widget.valid =
+      widget.valid && hideAfterActionChecker(condition.hideAfterAction, widget);
+  }
 
-    if (condition.urlContains) {
-      widget.valid = widget.valid && urlChecker(condition.urlContains);
-    }
+  if (condition.urlContains) {
+    widget.valid = widget.valid && urlChecker(condition.urlContains);
+  }
 
-    if (condition.metaContains) {
-      widget.valid = widget.valid && metaChecker(condition.metaContains);
-    }
+  if (condition.metaContains) {
+    widget.valid = widget.valid && metaChecker(condition.metaContains);
+  }
 
-    widget.valid = widget.valid && condition.showOnInit;
+  widget.valid = widget.valid && condition.showOnInit;
 
-    if (condition.impressions) {
-      widget.valid =
-        widget.valid && impressionsChecker(condition.impressions, widget);
-    }
+  if (condition.impressions) {
+    widget.valid =
+      widget.valid && impressionsChecker(condition.impressions, widget);
+  }
 
-    if (
-      typeof condition.priority !== 'undefined' &&
-      widget.valid &&
-      widgetTracker.prioritizedWidgets.indexOf(widget) === -1
-    ) {
-      widgetTracker.prioritizedWidgets.push(widget);
-      return;
-    }
+  // if it's valid at this point, add it to the priority list
+  if (widget.valid && options && options.priority === 'ordered') {
```

Comment:
we should open a ticket for this as its a bigger thing but we should work towards moving all these strings into constants so that 1) its easier to manage and 2) we can do more validation.